src/ButtonGroup/index.js

```diff
@@ -0,0 +1,178 @@
+import React, { PureComponent, Children, cloneElement } from 'react';
+import PropTypes from 'prop-types';
+import styled, { css } from 'styled-components';
+import { setLightness, transparentize } from 'polished';
+
+import Button from '../Button';
+
+const { bool, func, node, string } = PropTypes;
+
+/**
+ * Button Group
+ *
+ * This component is in charge of displaying
+ * a button group
+ *
+ * @param {node} children // Anything that can be rendered: numbers, strings, elements or an array (or fragment).
+ * @param {string} className // Add a text aside in the select next the selected value.
+ * @param {boolean} primary // whether the button is a primary button or not.
+ * @param {boolean} secondary // whether the button is a secondary button or not.
+ * @param {boolean} success // whether the button is a success button or not.
+ * @param {boolean} failure // whether the button is a failure button or not.
+ * @param {boolean} fluid // whether the button is fluid or not.
+ * @param {boolean} big // whether the button is big or not.
+ * @param {boolean} small // whether the button is small or not.
+ * @param {boolean} disabled // whether the button is disabled or not.
+ * @param {function} onChange // Callback function called when a button is clicked.
+ * @param {string} defaultActiveButton // Specifies the initial state: witch button is selected.
+ *
+ * @return {jsx}
+ */
+class ButtonGroup extends PureComponent {
+  /**
+   * PropTypes Validation.
+   */
+  static propTypes = {
+    children: node,
+    className: string,
+    primary: bool,
+    secondary: bool,
+    success: bool,
+    failure: bool,
+    onChange: func,
+    defaultActiveButton: string,
+  };
+
+  /**
+   * Default props.
+   */
+  static defaultProps = {
+    children: null,
+    className: '',
+    primary: false,
+    secondary: false,
+    success: false,
+    failure: false,
+    onChange: () => {},
+    defaultActiveButton: '',
+  };
+
+  /** Internal state. */
+  state = {
+    activeButton: this.props.defaultActiveButton, // eslint-disable-line react/destructuring-assignment
+  };
+
+  /**
+   * updateActiveButton.
+   * Tell to the store witch button is activated
+   * @param {string} name
+   */
+  updateActiveButton = name => {
+    const { onChange } = this.props;
+
+    this.setState({ activeButton: name });
+    onChange && onChange(name);
+  };
+
+  /**
+   * Renders the component.
+   *
+   * @return {jsx}
+   */
+  render() {
+    const { children, className, ...rest } = this.props;
+    const { activeButton } = this.state;
+
+    return (
+      <div role="group" className={className}>
+        {Children.map(children, child =>
+          cloneElement(child, {
+            onClick: () => this.updateActiveButton(child.props.name),
```

Comment:
Hmm... You completely erase the previously set `onClick` prop of the button. Couldn't you juste wrap it with a function that first calls the `updateActiveButton` of the `ButtonGroup` and then calls the initially set `onClick` prop of the button?